pkg/acsengine/defaults.go

```diff
@@ -313,6 +313,24 @@ var (
 			},
 		},
 	}
+
+	// DefaultContainerMonitoringAddonsConfig is the default container monitoring Kubernetes addon Config
+	DefaultContainerMonitoringAddonsConfig = api.KubernetesAddon{
+		Name:    ContainerMonitoringAddonName,
+		Enabled: helpers.PointerToBool(api.DefaultContainerMonitoringAddonEnabled),
+		Config: map[string]string{
+			"omsAgentVersion":       "1.6.0-42",
+			"dockerProviderVersion": "2.0.0-2",
+			"workspaceGuid":         "c5905df1-b3b9-42b9-acf0-14a4c4ef028c",
+			"workspaceKey":          "b4xMPEns/5Oo61hQbRLJQoPhbmAHdBn2eDjkCbRzgEISiKi9m1CR+093sczO9E8iA5w1EyGVCVkwsnuWt1MD9w==",
```

Comment:
we should probably not have a hardcoded key in the code

Reply:
@rashmichandrashekar this question is for you :)

Reply:
Sorry missed this :)
That is a good point. Is there a way for the acs-engine to talk to a secret store or key vault so that we can put these values there?

Reply:
On a second thought - since we will be testing this out as an addon, is there a way the cluster can take these as input parameters and pass them in during testing?

Reply:
These params are user-configurable, e.g., in your cluster api model:

```
"addons": [
          {
            "name": "container-monitoring",
            "enabled": true,
            "config": {
                "omsAgentVersion": "1.6.0-42",
                "dockerProviderVersion": "2.0.0-2",
		"workspaceGuid": "c5905df1-b3b9-42b9-acf0-14a4c4ef028c",
		"workspaceKey": "b4xMPEns/5Oo61hQbRLJQoPhbmAHdBn2eDjkCbRzgEISiKi9m1CR+093sczO9E8iA5w1EyGVCVkwsnuWt1MD9w=="
            }
            "containers": [
              {
                "name": "omsagent",
                "cpuRequests": "50m",
                "memoryRequests": "150Mi",
                "cpuLimits": "50m",
                "memoryLimits": "150Mi"
              }
            ]
          }
```

Reply:
okay. In that case we can update this to be passed in from the cluster api model.

Reply:
ACK. Should I remove the hardcoded "workspaceGuid" and "workspaceKey" default values, then?

Reply:
Yes please. Thank you.